src/lib/components/Geometry.svelte

```diff
@@ -33,95 +33,111 @@
 		...rest
 	}: Props = $props()
 
-	const type = $derived(geometry?.case)
+	const type = $derived(geometry?.geometryType?.case)
+	const color = $derived(overrideColor ?? metadata.color ?? colors.default)
+
+	const group = new Group()
 	const mesh = $derived.by(() => {
-		const object3d = type === undefined ? new Object3D() : new Mesh()
+		if (type === undefined) {
+			return
+		}
+		const result = new Mesh()
 
 		if (type === 'mesh' || type === 'points' || type === 'line') {
-			object3d.raycast = meshBounds
+			result.raycast = meshBounds
 		}
 
-		return object3d
+		return result
+	})
+
+	$effect.pre(() => {
+		if (geometry?.center && mesh) {
+			poseToObject3d(geometry.center, mesh)
+		}
 	})
 
 	$effect.pre(() => {
-		poseToObject3d(pose, mesh)
+		poseToObject3d(pose, group)
 	})
 
 	let geo = $state.raw<BufferGeometry>()
 
 	const oncreate = (ref: BufferGeometry) => {
 		geo = ref
 	}
-
-	const color = $derived(overrideColor ?? metadata.color ?? colors.default)
 </script>
 
 <T
```

Comment:
We introduce a group that represents the frame's offset, the child mesh now accepts the `center` offset